docs/eks-v3-migration.md

```diff
@@ -0,0 +1,150 @@
+  You can now also configure custom user data, giving more flexibility to users who need to run specific scripts or commands on node startup.  
+    
+- **Default Instance Type Update: t3.medium**  
+  The default instance type for node groups has been updated from `t2.medium` to `t3.medium`. The `t3.medium` instance type provides several advantages:  
+    
+  - **Higher Availability**: `t2.medium` is an older instance type that is being phased out in AWS regions, which can cause capacity issues during scaling. The newer `t3.medium` instance type offers better availability.
+  - **Improved Cost Efficiency**: `t3.medium` is marginally cheaper than `t2.medium` ($0.0416 vs $0.0464 per hour), providing better performance at a lower cost.
+
+
+- **New `operatingSystem` Property for Node Groups**  
+  A new `operatingSystem` property has been introduced for node groups, giving users the flexibility to choose which OS their nodes should run. Supported options include Amazon Linux 2023 (AL2023) and Bottlerocket, allowing you to select the OS that best fits your workloads and compliance needs.  
+    
+  - **`RECOMMENDED` Default Setting**: By default, the `RECOMMENDED` option will be used, which ensures your nodes are always running the OS recommended by AWS for optimal security and performance. This default setting will be kept in sync with AWS’s updates.
+
+
+- **EKS Cluster Addon Management: `vpc-cni`, `coredns` and `kube-proxy`**
+  The EKS cluster components `vpc-cni`, `coredns` and `kube-proxy` are now configured as EKS managed addons. This change eliminates the dependency on `kubectl` for managing these components and automates their updates, ensuring that clusters stay up to date without manual intervention.
+  This also simplifies management of clusters with private API endpoints because those components are managed via AWS APIs.
+  If you want to manage those EKS components yourself, you can do so by disabling them in the cluster component.
+    
+- **Support for Cluster Autoscaler and External Scaling**  
+  Pulumi EKS v3 introduces support for integrating with the Kubernetes cluster-autoscaler, which allows your cluster to scale dynamically based on resource needs. A new `ignoreScalingChanges` parameter has been added for node groups. This parameter allows Pulumi to ignore external scaling changes (e.g., changes to the desired size made by the cluster-autoscaler).
+    
+- **Cluster Version Tracking in `ManagedNodeGroup`**  
+  `ManagedNodeGroup` now automatically tracks the EKS cluster version, ensuring smoother version upgrades and preventing version mismatches. Since EKS only allows one minor version skew between the cluster control plane and the nodes, this automatic tracking helps prevent upgrade failures due to version incompatibility.
+
+# Migration
+
+## VPC CNI Component changes
+
+The VPC CNI cluster component is now configured as an EKS addon as mentioned in the “New Features” section above. This brings the following changes:
```

Comment:
Should it also mention this diff?

```
+      ├─ eks:index:VpcCniAddon        cluster-vpc-cni                  create     
+      │  └─ aws:eks:Addon             cluster-vpc-cni                  create     
-      ├─ eks:index:VpcCni             cluster-vpc-cni                  delete
 ```

Reply:
Yeah, I can add a sentence about this!

Reply:
@mikhailshilkov  https://github.com/pulumi/pulumi-eks/pull/1428